src/RootCloseWrapper.js

```diff
@@ -4,6 +4,14 @@ import EventListener from './utils/EventListener';
 
 // TODO: Merge this logic with dropdown logic once #526 is done.
 
+const clickWasInside = Symbol('clickWasInside');
+
```

Comment:
This is a really clever approach and seems like a better option than stopImmediatePropagation, since the clicks will still be passed through. One thing though is that using a symbol, while probably better, introduces a lot of extra code to the project to polyfill them. We should look at how much code is introduced by babel/corejs for symbols just so we are all aware of the additional bytes.